src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/DiaryPage.java

```diff
@@ -13,12 +17,56 @@ public class DiaryPage extends DiaryPageBase {
     @ExtendedFindBy(accessibilityId = "View Nutrition")
     private ExtendedWebElement viewNutrition;
 
+    @ExtendedFindBy(accessibilityId = "Edit Diary")
+    private ExtendedWebElement editDiaryButton;
+
+    @FindBy(id = "com.myfitnesspal.android:id/goal")
+    private ExtendedWebElement goalCalories;
+
+    @FindBy(id = "com.myfitnesspal.android:id/remaining_diary")
+    private ExtendedWebElement remainCalories;
+
     public DiaryPage(WebDriver driver) {
         super(driver);
     }
 
+    @Override
+    public boolean isDiaryClean() {
+        return !editDiaryButton.isPresent(1);
+    }
+
+    @Override
+    public void clearDiary() {
+        if(!isPageOpened())
+        {
+            CommonPageBase commonPage = initPage(getDriver(), CommonPageBase.class);
+            commonPage.clickBottomNavigatorButton(BottomNavigatorButtons.DIARY);
```

Comment:
wrong logic - we shoul call this method ONLY from Diary page, all this structure should be deleted